### PR TITLE
Gui testcases issue 2642

### PR DESCRIPTION
**Pull request by maryvictol**
This PR adds testcases for Restrict a count of the instances to launch #2642

### CODE REVIEW

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_2.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- Non-admin user is a member of `<User_Group1>`
+- Perform [2642_1](2642_1.md) case
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the admin user from the prerequisites | |
+| 2 | Open the **Settings** page | |
+| 3 | Open the **USER MANAGEMENT** tab | |
+| 4 | Click the **Groups** tab | |
+| 5 | Find `User_Group` group and open it to edit | |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_2.md]: Fix to `User_Group1`

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+| 19 | Open the **Runs** page | |
+| 20 | Click the **Rerun** hyperlink for run paused at step 4 of Preparations | Error message `Launch of new jobs is restricted as [<non-admin user>] user will exceed [<user-contextual-limit>] runs limit [3]` appears, where <br> `<non-admin user>` is the non-admin user from the prerequisites, <br> `3` is the value specified at step 6 |
+| 21 | Click the run launched at step 12 | |
+| 22 | Wait until the **SSH** hyperlink appears | |
+| 23 | Click the **SSH** hyperlink | |
+| 24 | In the opened tab, enter and perform the command: `pipe run -di <docker image> -y`, <br> where `<docker image>` is the test tool name with path | <li> Job isn't started <li> The command output is `Launch of new jobs is restricted as [<non-admin user>] user will exceed [<User_Group1>] runs limit [2]` appears, where <br> `<non-admin user>` is the non-admin user from the prerequisites, <br> `2` is the value specified at step 6 |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md]: I suppose that error message will be the same as at step 20

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+| 25 | Enter and perform the command: `pipe users instances --verbose` | Table with all running instances restrictions specified for the current user |
+
+**After:**
+- Stop runs launched at steps 12, 13
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md:50]: Add:
"- Terminate run that was paused at step 4 of Preparations"

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+6. In the *Launch options* section input value `3` into the *Allowed instance max count* field
+7. Click **OK** button
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the non-admin user from the prerequisites | |
+| 2 | Open the **Tools** page | |
+| 3 | Select the test tool | |
+| 4 | Launch a tool with default settings | |
+| 5 | Repeat steps 2-3 | |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md:24]: Fix to `Repeat step 2` or vice versa remove step 6

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+| 14 | Click **Launch** button | Error message `Launch of new jobs is restricted as [<non-admin user>] user will exceed [<user-contextual-limit>] runs limit [3]` appears, where <br> `<non-admin user>` is the non-admin user from the prerequisites, <br> `3` is the value specified at step 6 of the Preparations |
+| 15 | Repeat steps 9-10 | |
+| 16 | Set value `1` into ***Child nodes:*** field | Error message disappeared | |
+| 17 | Launch tool | |
+| 18 | Select the test tool | |
+| 19 | Launch a tool with default settings | Error message `Launch of new jobs is restricted as [<non-admin user>] user will exceed [<user-contextual-limit>] runs limit [3]` appears |
+| 20 | Click the run launched at step 4 | |
+| 21 | Wait until the **SSH** hyperlink appears | |
+| 22 | Click the **SSH** hyperlink | |
+| 23 | Enter and perform the command: `pipe users instances` | Response contains follow info: <li> `Active runs detected for a user: [<non-admin user>: 3]` <li> `The following restriction applied on runs launching: [<non-admin user>: 3]` |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md]: I've tried to perform a test case, and the response I've got - `Active runs detected for a user: [<non-admin user>: 2] ...`

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+- Non-admin user
+- Perform [2642_5](2642_5.md) case
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+|   1   | Login as the admin user from the prerequisites | |
+|   2   | Open the **Tools** page | |
+|   3   | Select the test tool | |
+|   4   | Open **Settings** tab | |
+|   5   | Expand the **Exec environment** section | |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md]: Fix to `Execution environment`

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+|   6   | Click the **Configure cluster** control | |
+|   7   | Click the **Cluster** tab | |
+|   8   | Set value `2` into ***Child nodes:*** field | |
+|   9   | Click the **OK** button | |
+|  10   | Click **Save** button | |
+|  11   | Login as the non-admin user from the prerequisites | |
+|  12   | Open the **Tools** page | |
+|  13   | Select the test tool | |
+|  14   | Launch a tool with default settings | **Launch** pop-up that contains error message `You have exceeded maximum number of running jobs (3).` appears |
+|  15   | Click **Launch** button | Error message `Launch of new jobs is restricted as [<non-admin user>] user will exceed [<user-contextual-limit>] runs limit [3]` appears, where <br> `<non-admin user>` is the non-admin user from the prerequisites, <br> `3` is the value specified at step 6 of the Preparations |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md]: Add:
`... at step 6 of the Preparations of the [2642_5](2642_5.md) case`

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+6. In the *Launch options* section input value `3` into the *Allowed instance max count* field
+7. Click **OK** button
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as the non-admin user from the prerequisites | |
+| 2 | Open the **Tools** page | |
+| 3 | Select the test tool | |
+| 4 | Launch a tool with default settings | |
+| 5 | Repeat steps 2-3 | |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md:24]: Fix to `Repeat step 2` or vice versa remove step 6

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+**After:**
+- Stop runs launched at steps 4, 17 of [2642_5](2642_5.md) case
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md:29]: Add:
`- Remove cluster option configured for the test tool at steps 8-10`

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+| 14 | Set value `1` into ***Default child nodes:*** field | Warning message `Your cluster configuration may exceed the maximum number of running jobs. There are 1 jobs running out of 3.` is shown on the ***Configure cluster*** pop-up |
+| 15 | Click the **OK** button | Warning message `Your cluster configuration may exceed the maximum number of running jobs. There are 1 jobs running out of 3.` is shown in the ***Exec environment*** section |
+| 16 | Click **Launch** button | **Launch** pop-up that contains warning message `Your cluster configuration may exceed the maximum number of running jobs. There are 1 jobs running out of 3.` appears | 
+| 17 | Click **Launch** button | |
+| 18 | Open the just-launched run | |
+| 19 | Wait until the label "Nested runs" appears | Only 1 child run is created |
+| 20 | Wait until the **SSH** hyperlink appears | |
+| 22 | Click the **SSH** hyperlink | |
+| 23 | Enter and perform the command: `pipe users instances` | Response contains follow info: <li> `Active runs detected for a user: [<non-admin user>: 3]` |
```
Comment (NShaforostov) [inline docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md]: I've tried to perform a test case, and the response I've got - `Active runs detected for a user: [<non-admin user>: 2] ...`

**Review by NShaforostov — APPROVED** [review]
Fix minor notes (see comments) before the merge, please.

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_2.md (hunk 1 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_2.md]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md (hunk 2 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md (hunk 3 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_4.md:50]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md (hunk 4 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md:24]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md (hunk 6 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md (hunk 7 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md (hunk 8 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md:24]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md (hunk 9 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_6.md:29]: Done

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md (hunk 10 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_7.md]: This's result the bug described in https://github.com/epam/cloud-pipeline/issues/2642#issuecomment-1171475127

File: docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md (hunk 5 above)
Comment (maryvictol) [reply docs/testcases/UI/Run_tools_in_sandbox/Launch_parameters/2642/2642_5.md]: This's result the bug described in https://github.com/epam/cloud-pipeline/issues/2642#issuecomment-1171475127

**Pull request merged**